diesel/src/pg/connection/mod.rs

```diff
@@ -180,7 +180,7 @@ impl crate::r2d2::R2D2Connection for PgConnection {
         self.transaction_state
             .status
             .transaction_depth()
-            .map(|d| d.is_none())
+            .map(|d| d.is_some())
             .unwrap_or(true)
```

Comment:
Curious how come this couldn't be something like:
```
match self.transaction_state.status {
    TransactionManagerStatus::Valid(_) => false,
    TransactionManagerStatus::InError => true,
}
```

instead?

Reply:
We also want to check the transaction depth - if we failed/forgot to rollback the current transaction (e.g. in case of panic) before releasing to the pool, any new use of the connection would be within a transaction while it would not expect to.

So it looks like that piece of documentation you linked should be the one that gets updated.

Side note: I think this function may be easier to read (mistake would have been harder to make) if it was written in the following form:
```rust
match self.transaction_state.status.transaction_depth() {
    Ok(None) => false,
    Err(_) | Ok(Some(_)) => true,
}
```

Reply:
Ahh I guess I was confused because I thought there may be a legitimate reason for the transaction depth to be non-0 at the point when the connection gets put back. But now that I think about it, that's not likely to happen since the flow would _usually_ be (a) task acquires connection from pool, (b) does all operations including transactions, (c) hopefully transactions complete successfully at which point the transaction depth returns to 0 or, if they don't, hopefully rollbacks reset transaction depth appropriately to 0, (d) release back to pool. But as you say in edge cases such as panics, then yeah the transaction depth not being 0 at point (d) entails that the connection is "dirty" so to speak.

Thanks for the note! And yeah the `match` clause you've provided I think seems quite nice :) Perhaps the documentation can be updated as part of this PR, but if not I'll be happy to do it since I initially wrote it myself 😅